src/lib.rs

```diff
@@ -3922,6 +3921,16 @@ impl Build {
             ))
         }
     }
+    fn is_wasi_target(target: &str) -> bool {
+        const TARGETS: [&'static str; 5] = [
+            "wasm32-wasi",
+            "wasm32-wasip1",
+            "wasm32-wasip1-threads",
+            "wasm32-wasip2",
+            "wasm32-wasi-threads",
+        ];
+        return TARGETS.contains(&target);
```

Comment:
Why are we listing each target? Would `target.contains("wasi")` not be more forwards-compatible?

Reply:
Maybe, it was mostly to ensure if they had a target supporting exceptions or something like that to handle them separately, but maybe its overkill